sdk/spring/spring-cloud-azure-autoconfigure/src/test/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsMessagingAutoConfigurationTests.java

```diff
@@ -71,4 +78,67 @@ void connectionInfoAndCheckpointStoreProvidedShouldConfigure() {
             });
     }
 
+    @Test
+    void withoutObjectMapperShouldNotConfigure() {
+        this.contextRunner
+            .withClassLoader(new FilteredClassLoader(ObjectMapper.class))
+            .withConfiguration(AutoConfigurations.of(JacksonAutoConfiguration.class))
+            .withPropertyValues(
+            "spring.cloud.azure.eventhubs.connection-string=" + String.format(CONNECTION_STRING_FORMAT, "test-namespace")
+        )
+            .withUserConfiguration(AzureEventHubsPropertiesTestConfiguration.class)
+            .run(context -> assertThatIllegalStateException());
+    }
+
+    @Test
+    void withIsolatedObjectMapper() {
+        this.contextRunner
+            .withPropertyValues("spring.cloud.azure.eventhubs.connection-string=" + String.format(CONNECTION_STRING_FORMAT, "test-namespace"))
+            .withUserConfiguration(AzureEventHubsPropertiesTestConfiguration.class)
+            .withConfiguration(AutoConfigurations.of(JacksonAutoConfiguration.class))
+            .run(context -> {
+                assertNotNull(context.getBean("defaultEventHubsMessageConverter"));
+                assertThat(context).hasSingleBean(EventHubsMessageConverter.class);
+                assertThat(context).doesNotHaveBean("eventHubsMessageConverter");
+            });
+    }
+
+    @Test
+    void withNonIsolatedObjectMapper() {
+        this.contextRunner
+            .withPropertyValues("spring.cloud.azure.eventhubs.connection-string=" + String.format(CONNECTION_STRING_FORMAT, "test-namespace"),
+                "spring.cloud.azure.message-converter.isolated-object-mapper=false")
+            .withUserConfiguration(AzureEventHubsPropertiesTestConfiguration.class)
+            .withConfiguration(AutoConfigurations.of(JacksonAutoConfiguration.class))
+            .run(context -> {
+                assertNotNull(context.getBean("eventHubsMessageConverter"));
+                assertThat(context).hasSingleBean(EventHubsMessageConverter.class);
+                assertThat(context).doesNotHaveBean("defaultEventHubsMessageConverter");
+            });
+    }
+
+    @Test
+    void withUserProvidedObjectMapper() {
+        this.contextRunner
+            .withPropertyValues("spring.cloud.azure.eventhubs.connection-string=" + String.format(CONNECTION_STRING_FORMAT, "test-namespace"),
+                "spring.cloud.azure.message-converter.isolated-object-mapper=false")
+            .withUserConfiguration(AzureEventHubsPropertiesTestConfiguration.class)
+            .withUserConfiguration(UserCustomizeObjectMapperConfiguration.class)
```

Comment:
How about using `org.springframework.boot.test.context.runner.AbstractApplicationContextRunner#withBean(java.lang.String, java.lang.Class<T>, java.util.function.Supplier<T>, org.springframework.beans.factory.config.BeanDefinitionCustomizer...)` instead of `withUserConfiguration` and another configuration inner class?